substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixJavaNetSubstitutions.java

```diff
@@ -119,84 +419,1339 @@ protected int peekData(DatagramPacket p) throws IOException {
+                //throw new SocketException(PosixUtils.lastErrorString("Error getting socket option"));
+                //  1937                  return -1;
+                return -1;
+            }
+            //  1938              }
+            //  1939          return (jint)ttl;
+            return ttl_Pointer.read();
+        } else {
+            //  1941  #endif /* AF_INET6 */
+            //  1943              u_char ttl = 0;
+            CCharPointer ttl_Pointer = StackValue.get(CCharPointer.class);
+            ttl_Pointer.write((byte) 0);
+
+            //  1944              int len = sizeof(ttl);
+            CIntPointer len_Pointer = StackValue.get(CIntPointer.class);
+            len_Pointer.write(SizeOf.get(CCharPointer.class));
+
+            //  1945              if (JVM_GetSockOpt(fd, IPPROTO_IP, IP_MULTICAST_TTL,
+            //  1946                                 (char*)&ttl, &len) < 0) {
+            if (Socket.getsockopt(fd, NetinetIn.IPPROTO_IP(), NetinetIn.IP_MULTICAST_TTL(), ttl_Pointer, len_Pointer) < 0) {
+                //  1947                  NET_ThrowByNameWithLastError(env, JNU_JAVANETPKG "SocketException",
+                //  1948                                 "Error getting socket option");
+                throw new SocketException(PosixUtils.lastErrorString("Error getting socket option"));
+                //  1949                  return -1;
+            }
+            //  1951              return (jint)ttl;
+
+            return ttl_Pointer.read();
+        }
     }
 
+    //   248  /*
+    //   249   * Class:     java_net_PlainDatagramSocketImpl
+    //   250   * Method:    connect0
+    //   251   * Signature: (Ljava/net/InetAddress;I)V
+    //   252   */
+    //   253  JNIEXPORT void JNICALL
+    //   254  Java_java_net_PlainDatagramSocketImpl_connect0(JNIEnv *env, jobject this,
+    //   255                                                 jobject address, jint port)
     @Substitute
     @SuppressWarnings({"static-method", "unused"})
-    protected void connect0(InetAddress address, int port) {
-        throw VMError.unsupportedFeature("Unimplemented: java.net.PlainDatagramSocketImpl.connect0(InetAddress, int)");
+    protected void connect0(InetAddress address, int port) throws SocketException {
+        /* Do not re-format commented out code: @formatter:off */
+        //   256      /* The object's field */
+        //   257      jobject fdObj = (*env)->GetObjectField(env, this, pdsi_fdID);
+        FileDescriptor fdObj = Util_java_net_DatagramSocketImpl.as_Target_java_net_DatagramSocketImpl(this).fd;
+        //   258      /* The fdObj'fd */
+        //   259      jint fd;
+        int fd;
+        //   260      /* The packetAddress address, family and port */
+        //   261      SOCKADDR rmtaddr;
+        Socket.sockaddr rmtaddr = StackValue.get(Socket.sockaddr.class);
+
+        //   262      int len = 0;
+        CIntPointer len_Pointer = StackValue.get(CIntPointer.class);
```

Comment:
`len_Pointer.write(0)`.  Maybe unnecessary, but I want to read this line of translation and know that it is correct in and of itself.